src/trio/_core/_tests/test_exceptiongroup_gc.py

```diff
@@ -0,0 +1,101 @@
+from __future__ import annotations
+
+import gc
+import sys
+from traceback import extract_tb
+from typing import TYPE_CHECKING, Callable, NoReturn
+
+import pytest
+
+from .._concat_tb import concat_tb
+
+if TYPE_CHECKING:
+    from types import TracebackType
+
+if sys.version_info < (3, 11):
+    from exceptiongroup import ExceptionGroup
+
+
+def raiser1() -> NoReturn:
+    raiser1_2()
+
+
+def raiser1_2() -> NoReturn:
+    raiser1_3()
+
+
+def raiser1_3() -> NoReturn:
+    raise ValueError("raiser1_string")
+
+
+def raiser2() -> NoReturn:
+    raiser2_2()
+
+
+def raiser2_2() -> NoReturn:
+    raise KeyError("raiser2_string")
+
+
+def get_exc(raiser: Callable[[], NoReturn]) -> Exception:
+    try:
+        raiser()
+    except Exception as exc:
+        return exc
+    raise AssertionError("raiser should always raise")  # pragma: no cover
+
+
+def get_tb(raiser: Callable[[], NoReturn]) -> TracebackType | None:
+    return get_exc(raiser).__traceback__
+
+
+def test_concat_tb() -> None:
+    tb1 = get_tb(raiser1)
+    tb2 = get_tb(raiser2)
+
+    # These return a list of (filename, lineno, fn name, text) tuples
+    # https://docs.python.org/3/library/traceback.html#traceback.extract_tb
+    entries1 = extract_tb(tb1)
+    entries2 = extract_tb(tb2)
+
+    tb12 = concat_tb(tb1, tb2)
+    assert extract_tb(tb12) == entries1 + entries2
+
+    tb21 = concat_tb(tb2, tb1)
+    assert extract_tb(tb21) == entries2 + entries1
+
+    # Check degenerate cases
+    assert extract_tb(concat_tb(None, tb1)) == entries1
+    assert extract_tb(concat_tb(tb1, None)) == entries1
+    assert concat_tb(None, None) is None
+
+    # Make sure the original tracebacks didn't get mutated by mistake
+    assert extract_tb(get_tb(raiser1)) == entries1
+    assert extract_tb(get_tb(raiser2)) == entries2
+
+
+# Unclear if this can still fail, removing the `del` from _concat_tb.copy_tb does not seem
+# to trigger it (on a platform where the `del` is executed)
```

Comment:
Huh, it might be a good idea to remove this test then and add a `TODO` to that `del` that it isn't tested for.

It seems like a good idea to try to remove unnecessary `del`s if things have improved!

Reply:
see https://github.com/python-trio/trio/pull/2891#issuecomment-1831844141 on removing unnecessary `del`s.
But `copy_tb` is possibly an exception since it no longer uses `MultiError`.